isclib.go

```diff
@@ -87,10 +107,15 @@ func SetExecuteTemporaryDirectory(path string) {
 	executeTemporaryDirectory = path
 }
 
-// ISCExists returns a boolean which is true when an ISC product or Caché instance exists on this system.
+// ISCExists returns a boolean which is true when an ISC product instance exists on this system.
 func ISCExists() bool {
```

Comment:
I would probably get rid of all of these various `*Exists` functions and just have a single function similar to the following.

```go

type ISCProduct int

const (
  ProductNone ISCProduct = iota
  ProductIris
  ProductEnsemble
  ProductCache
)

func Product() (ISCProduct) {
  // Determine if any product exists and if so return which one
  // Additionally, you could cache this result easily so that we don't have to repeatedly do this check.
}
```
If the `bool` return is desired for whatever reason, you could do this as well...
```go
func Exists() (bool, ISCProduct) {
  p := Product()
  return p != ProductNone, p
}
```

That being said, the way it is currently written you could have both Iris and Caché/Ensemble on the system. Is that the intent? If it is, I would modify the above to use bitmasks and change the name of `Product()` to `Products()`.